pydantic/functional_serializers.py

```diff
@@ -177,55 +177,52 @@ def __get_pydantic_core_schema__(self, source_type: Any, handler: GetCoreSchemaH
 
 
 if TYPE_CHECKING:
-    _PartialClsOrStaticMethod: TypeAlias = Union[classmethod[Any, Any, Any], staticmethod[Any, Any], partialmethod[Any]]
-    _PlainSerializationFunction = Union[_core_schema.SerializerFunction, _PartialClsOrStaticMethod]
-    _WrapSerializationFunction = Union[_core_schema.WrapSerializerFunction, _PartialClsOrStaticMethod]
-    _PlainSerializeMethodType = TypeVar('_PlainSerializeMethodType', bound=_PlainSerializationFunction)
-    _WrapSerializeMethodType = TypeVar('_WrapSerializeMethodType', bound=_WrapSerializationFunction)
+    _Partial: TypeAlias = 'partial[Any] | partialmethod[Any]'
 
+    AnyFieldPlainSerializer: TypeAlias = 'core_schema.SerializerFunction | _Partial'
+    """A field serializer method or function in `plain` mode."""
```

Comment:
I really like these changes overall, but I'm a bit hesitant regarding the use of `Any` here as a prefix for these types. Perhaps this might get confusing with the type `Any`... let's brainstorm some alternatives - I don't think `Either` is good...